packages/backend/src/server/routers/campaign.ts

```diff
@@ -3,11 +3,15 @@ import { isKyc } from '../session'
 import { publicProcedure, router } from '../trpc'
 import { prisma } from '../prisma'
 // import { S3Client, PutObjectCommand } from '@aws-sdk/client-s3'
+import { getChain } from '@icecreamswap/constants/src/chains'
+import { Contract, providers } from 'ethers'
+import campaignFactoryAbi from '@passive-income/launchpad-contracts/abi/contracts/PSIPadCampaignFactory.sol/PSIPadCampaignFactory.json'
 
 export const campaignRouter = router({
   add: publicProcedure
     .input(
       z.object({
+        user: z.string().length(42),
```

Comment:
don't use an invalidated user address, use the user address from the signed ctx.session.user object.